src/python/pants/backend/python/dependency_inference/rules.py

```diff
@@ -264,6 +266,57 @@ def _resolve_single_asset(filepath) -> ImportResolveResult:
     return {filepath: _resolve_single_asset(filepath) for filepath in assets}
 
 
+@rule_helper
+async def _disambiguate_or_mark_unowned(
+    address: Address,
+    resolve_results: dict[str, ImportResolveResult],
+    explicitly_provided_deps: ExplicitlyProvidedDependencies,
+) -> dict[str, ImportResolveResult]:
+    """Use dependency rules to disambiguate imports based on Address."""
+    ambiguous_results = {
+        import_name: result
+        for import_name, result in resolve_results
+        if result.status == ImportOwnerStatus.ambiguous
+    }
+    if not ambiguous_results:
+        return resolve_results
+
+    dependencies_rule_actions = await MuliGet(
+        Get(
+            DependenciesRuleAction,
+            DependenciesRuleActionRequest(
+                address=address,
+                dependencies=result.address,
+                description_of_origin=f"get dependency rules for {address} to import {import_name}",
+            ),
+        ) for import_name, result in ambiguous_results.items()
+    )
+    dep_rules_enabled = dependencies_rule_actions[0].address is not None
```

Comment:
this is essentially `any(action.address is not None for action in dependencies_rule_actions)` that doesn't have to traverse the entire list, since we know (?) that they'll all be the same?

Reply:
Yes. They will all be the same. Would the `any()` variant make the purpose clearer here?